### DISCUSSION TITLE
substructure search issue - aromaticity perception

### DISCUSSION THREAD

**Discussion by YoavShamir5** [rdkit/rdkit#6965] [Q&A] [answered]
Hello all, I am trying to run a substructure search of smiles_b in smiles_a, but I get no matches. I realize this has to do with differences in aromaticity perception, but am not sure how to solve this. I should note I have the molecular data only in SMILES format.

smiles_a = "C=CC(=O)N1CCc2c(cccc2-c2c(C)nc(C(N)=O)c3[nH]c(C)c(C)c23)C1"
mol_a = Chem.MolFromSmiles(smiles_a)

smiles_b = ""CC1=NC(C(N)=O)=C2NC(C)=C(C)C2=C1C1=CC=CC2=C1CC[NH2+]C2"
mol_b = mol_b = Chem.MolFromSmiles(smiles_b)

matches = mol_a.GetSubstructMatches(mol_b) #yields no matches

**Comment by bp-kelley** [comment, marked as answer]
The issue is your charged nitrogen in smiles_b.

If we remove this, then you have the subgraph match you expect:
```
smiles_b = "CC1=NC(C(N)=O)=C2NC(C)=C(C)C2=C1C1=CC=CC2=C1CCNC2"
mol_b = Chem.MolFromSmiles(smiles_b)
mol_a.GetSubstructMatches(mol_b)
```

If this is a typical situation for you, you may want to use the MCS code (maximum common subgraph)

```
from rdkit.Chem import rdFMCS
res = rdFMCS.FindMCS([mol_a, mol_b])
res.smartsString
mol_a.GetSubstructMatch(res.queryMol)
```

![image](https://github.com/rdkit/rdkit/assets/9888416/8132ed0d-489c-4dd0-a964-071a50648024)

**Reply by YoavShamir5** [reply]
Fantastic, exactly what I was looking for, thanks!
